src/translate.py

```diff
@@ -4255,10 +4273,9 @@ def make_arg(offset: int, type: Type) -> PassedInArg:
         assert offset % 4 == 0
         return PassedInArg(offset, copied=False, stack_info=stack_info, type=type)
 
-    if typemap and function.name in typemap.functions:
-        fn_type = type_from_ctype(typemap.functions[function.name].type, typemap)
-        fn_decl_provided = True
-    else:
+    fn_type = universe.get_function_type(function.name)
```

Comment:
get_function_type -> get_known_function_type maybe? fn_decl_provided expects to use "is defined in context" as a heuristic for something, but `get_function_type` sounds like it may make function signatures up. (I do wonder if that knowledge of types can't move from GlobalInfo to TypeUniverse without too much pain.)